tests/kernel/context/src/main.c

```diff
@@ -127,7 +127,7 @@ static void isr_handler(const void *data)
 
 	switch (isr_info.command) {
 	case THREAD_SELF_CMD:
-		isr_info.data = (void *)k_current_get();
+		isr_info.data = (void *)k_sched_current_thread_query();
```

Comment:
Just to repeat: this here is what we can't do.  k_current_get() is the **correct API** for the test to use here.  It's not working, and that's a bug that the test has discovered.   If we're not going to fix the bug (and it sounds like you're not), you need to document that limitation somewhere and just accept that this test case won't pass.

And you don't need to filter the whole of tests/kernel/context.  Just a a `ztest_test_skip()` (when ARC && TLS) to the cases that hit this code in ISR context.

Reply:
Hi @andyross,
Regarding your advice I tried to cut off ```context``` and ```context_one_cpu``` tests for ARC.
To make compiller happy, I had to apply a lot of #ifdef conditions inside the code, that make the code ugly:
https://github.com/zephyrproject-rtos/zephyr/commit/cfc363619212a120b5e634bd12f7171f0efd6739

Another option - is apply runtime construction:
```
	if(IS_ENABLED(NO_TLS_IN_EXCEPTION)){
	    ztest_test_skip();
	}else{
	    ...
	}
```

Which one option is more applicable for current situation?